main.cpp

```diff
@@ -485,6 +486,21 @@ void ChannelStrip::acceptAudioInput(bool isConnected, bool isMono)
     std::cout << "How loud can it go? " << maxGainRange << "db" << std::endl;
 }
 
+float ChannelStrip::addChannel(int numChannels)
+{
+    int channel = 1;
+
+    for (int i = 0; i < numChannels; i++)
+    {
+        if (channel >= 1)
+        {
+            numChannels = channel += 1;
+            break;
+        }
+    }
+    return numChannels;
```

Comment:
The first time your loop runs, `channel` is already `>= 1`
That means the `if() { }` is entered, and that `break`; statement is hit, which exits the loop.
**that means it isn't looping**

do this, which will actually loop `numChannels` times.
```
float ChannelStrip::addChannel(int numChannels)
{
    int channel = 1;
    for (int i = 0; i < numChannels; i++)
    {
        ++channel; //add '1' to channel 'numChannels' times.  
        std::cout << "adding channel num: " << channel << std::endl;
    }
    return channel;
}
```
- [ ] revise